src/gtimelog/timelog.py

```diff
@@ -361,6 +364,26 @@ def totals(self, tag=None, filter_text=None):
                 total_work += duration
         return total_work, total_slacking
 
+    @classmethod
+    def _get_grouped_order_key(cls, sorted_by, sorted_tasks):
+        """
+        Returns a function suitable as key parameter for the ordered function
+
+        The parameter 'x' to be sorted is deemed a list item as returned by
+        TimeCollection.grouped_entries
+        """
+        # name is deemed unique as this function is used for grouped entries,
+        # hence sufficient to fully sort a list
+        if sorted_by == 'start-time':
+            return None  # hence sort by x
+        elif sorted_by == 'name':
+            return lambda x: x[1]
+        elif sorted_by == 'duration':  # return (duration, start-time, name)
+            return lambda x: (x[2], x[0], x[1])
+        elif sorted_by == 'task-list':
+            # name is also sent to order unknown entries in a stable way
```

Comment:
Hm.  My gut feeling says that start-time is better here (followed by duration for tie breaking), but I haven't thought about it very long.

Reply:
As my main ask for this whole thing was to have a sorting stable across weeks/days, the name, after the task-list itself, was for me the better choice (as commented).